### PR TITLE
Add minimal scenarios for boolean

### PR DESCRIPTION
**Pull request by hvub**
This PR adds scenarios for boolean operations. It is not necessarily complete and is not aimed to be. It is work done to get more clarity about the purpose to the mostly empty boolean features.

### CODE REVIEW

**Review by Mats-SX — APPROVED** [review]
This looks like a solid improvement. I'm noting that the `null` truth value is omitted from all scenarios. This may be OK given the scope of the work, but I think it's fair to note that the scenarios added are only partial in their coverage.

File: tck/features/expressions/boolean/Boolean1.feature [hunk 1]
```diff
@@ -29,3 +29,100 @@
 #encoding: utf-8
 
 Feature: Boolean1 - And logical operations
+
+  Scenario: [1] Conjunction of two truth values
```
[the remark this answers is not in the document]
Comment (hvub) [reply tck/features/expressions/boolean/Boolean1.feature:33]: No. You are absolutely right. Retrospectively I can only speculate, but apparently some forest for trees thing let complete forget about `null`. I will fix this.

File: tck/features/expressions/boolean/Boolean1.feature [hunk 2]
```diff
@@ -29,3 +29,100 @@
...
+      | true | false | false  | false  | false |
+    And no side effects
+
+  Scenario: [4] Conjunction is commutative
+    Given any graph
+    When executing query:
+      """
+      UNWIND [true, false] AS a
+      UNWIND [true, false] AS b
+      RETURN DISTINCT (a AND b) = (b AND a) AS result
```
[the remark this answers is not in the document]
Comment (hvub) [reply tck/features/expressions/boolean/Boolean1.feature:83]: I came from other scenarios for testing invariants. Those scenarios are not exhaustive (too many cases) but generate a large number of cases. You are right, here it is feasible and hence better to spell them out.

File: tck/features/expressions/boolean/Boolean1.feature [hunk 3]
```diff
@@ -29,3 +29,100 @@
...
+    And no side effects
+
+  Scenario: [5] Conjunction is associative
+    Given any graph
+    When executing query:
+      """
+      UNWIND [true, false] AS a
+      UNWIND [true, false] AS b
+      UNWIND [true, false] AS c
+      RETURN DISTINCT (a AND (b AND c)) = ((a AND b) AND c) AS result
```
[the remark this answers is not in the document]
Comment (hvub) [reply tck/features/expressions/boolean/Boolean1.feature:97]: I agree. Even with eight cases it is still feasible and hence better to spell them out.

File: tck/features/expressions/boolean/Boolean1.feature (hunk 1 above)
[the remark this answers is not in the document]
Comment (hvub) [reply tck/features/expressions/boolean/Boolean1.feature:33]: Done.

File: tck/features/expressions/boolean/Boolean1.feature (hunk 3 above)
[the remark this answers is not in the document]
Comment (hvub) [reply tck/features/expressions/boolean/Boolean1.feature]: Done.

File: tck/features/expressions/boolean/Boolean1.feature (hunk 2 above)
[the remark this answers is not in the document]
Comment (hvub) [reply tck/features/expressions/boolean/Boolean1.feature]: Done

**Comment by hvub** [comment]
@Mats-SX I have update the PR according to your comments. If you may have another look? Thx

**Comment by Mats-SX** [comment]
@hvub Very thorough! Looks great, and thanks for addressing the concerns. Extra points for removing the duplicate scenarios :)

**Comment by Mats-SX** [comment]
I added some per-commit comments btw, not sure why they don't show up here in the overview. Click the little "chat bubbles" on some of the commits to see them I guess

**Pull request merged**